vnext/Microsoft.ReactNative/Views/ReactRootControl.cpp

```diff
@@ -607,6 +608,53 @@ void ReactRootControl::ReloadViewHost() noexcept {
   }
 }
 
+void ReactRootControl::AttachBackHandlers(XamlView const &rootView) noexcept {
+  auto rootElement(rootView.as<winrt::UIElement>());
+  if (rootElement == nullptr) {
+    assert(false);
+    return;
+  }
+
+  winrt::Windows::UI::Core::SystemNavigationManager::GetForCurrentView().BackRequested(
+      [this](winrt::IInspectable const & /*sender*/, winrt::BackRequestedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+
+  // Handle keyboard "back" button press
+  winrt::KeyboardAccelerator goBack{};
+  goBack.Key(winrt::VirtualKey::GoBack);
+  goBack.Invoked(
+      [this](winrt::KeyboardAccelerator const & /*sender*/, winrt::KeyboardAcceleratorInvokedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+  rootElement.KeyboardAccelerators().Append(goBack);
+
+  // Handle Alt+Left keyboard shortcut
+  winrt::KeyboardAccelerator altLeft{};
+  altLeft.Key(winrt::VirtualKey::Left);
+  altLeft.Invoked(
+      [this](winrt::KeyboardAccelerator const & /*sender*/, winrt::KeyboardAcceleratorInvokedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+  rootElement.KeyboardAccelerators().Append(altLeft);
+  altLeft.Modifiers(winrt::VirtualKeyModifiers::Menu);
+}
+
+bool ReactRootControl::OnBackRequested() noexcept {
+  if (m_weakReactInstance.IsExpired()) {
```

Comment:
nit: checking to see whether a weakptr has died without acquiring a strong pointer is a bit of an antipatten, since it can do the wrong thing in multithreaded systems. E.g. if the react instance died after this but before we called GetStrongPtr().

Instead we should do something like:
```c++
if (auto reactInstance = m_weakReactInstance.GetStrongPtr()) {
// Stuff
}
```

This ensures the object is kept alive for the duration of its use.

I don't think it's a practical issue here, but is good to keep in mind as a general practice.